src/main/java/seedu/address/storage/JsonSerializableAddressBook.java

```diff
@@ -46,14 +50,47 @@ public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
         weddings.addAll(source.getWeddingList().stream().map(JsonAdaptedWedding::new).collect(Collectors.toList()));
     }
 
+    /**
+     * Checks if the wedding data integrity is maintained.
+     * All wedding jobs must have corresponding clients (non-zero ownWedding hashcodes).
+     *
+     * @throws IllegalValueException if wedding data integrity is violated.
+     */
+    private void validateWeddingIntegrity() throws IllegalValueException {
```

Comment:
this does not check that if ownWedding is altered

Reply:
should add something like this
```
public void validateWeddings(List<Wedding> weddings) {
        List<Integer> validWeddings = weddings.stream().map(Wedding::hashCode).toList();
```
then validate ownWeddings and weddingJobs based on the validWeddings